src/parquet/column/writer.cc

```diff
@@ -118,7 +119,13 @@ void ColumnWriter::AddDataPage() {
   DataPage page(
       uncompressed_data, num_buffered_values_, encoding_, Encoding::RLE, Encoding::RLE);
 
-  data_pages_.push_back(std::move(page));
+  // Write the page to OutputStream eagerly if there is no dictionary or
+  // if dictionary encoding has fallen back to PLAIN
```

Comment:
Sorry, my mistake, `data_pages_` is cleared after the fallback code path. Gerrit draft comments would be useful here